### ISSUE TITLE
Duration Calculator: Tests are failing

### ISSUE THREAD

**Issue by bsstoner** [duckduckgo/zeroclickinfo-goodies#3327]
Tests are currently failing:

```
3836 
3837  #  Failed test 'Deep: full ZCI object'
3838  #  at /opt/perlbrew/libs/perl-5.16.3@ddg/lib/perl5/DDG/Test/Goodie.pm line 54.
3839  # Compared $data->{"structured_answer"}{"data"}{"title"}
3840  # got : '4 weeks and 2 days'
3841  # expect : '1 month'
3842  # Looks like you failed 1 test of 1.
3843 
3844 #  Failed test 'Query: 4 weeks + 2 days'
3845 #  at /opt/perlbrew/libs/perl-5.16.3@ddg/lib/perl5/DDG/Test/Goodie.pm line 58.
3846 
3847  #  Failed test 'Deep: full ZCI object'
3848  #  at /opt/perlbrew/libs/perl-5.16.3@ddg/lib/perl5/DDG/Test/Goodie.pm line 54.
3849  # Compared $data->{"structured_answer"}{"data"}{"title"}
3850  # got : '-(1 month and 4 days)'
3851  # expect : '-(1 month and 3 days)'
3852  # Looks like you failed 1 test of 1.
3853 
3854 #  Failed test 'Query: 1 year - 400 days'
3855 #  at /opt/perlbrew/libs/perl-5.16.3@ddg/lib/perl5/DDG/Test/Goodie.pm line 58.
3856 # Looks like you failed 2 tests of 15.
3857 t/DurationCalculator.t ...........
```   

------
IA Page:  http://duck.co/ia/view/duration_calculator 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @sam09

**Comment by sam09** [comment]
The test are failing because of [this](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/DurationCalculator.pm#L40) line. We currently add  the resulting duration to the current time. So when you add 30 days to June it becomes 1 month but in July you would need 31 days for one month. That is why the tests failed when month changed.

I would suggest we use a fixed time instead of the current time. Any suggestions ? @gaulrobe @moollaza

**Comment by GuiltyDolphin** [comment]
@sam09 We should be using `Test::MockTime` in the tests to emulate a fixed time - if you want to go ahead and add that ping me and I'll get it merged ASAP.

**Comment by sam09** [comment]
I think we might need to fix that in the Goodie itself. We would want it to give consistent answers.

**Comment by sam09** [comment]
@GuiltyDolphin Check out #3329

**Comment by GuiltyDolphin** [comment]
@sam09 Hmm, yeah, months are a bit of a pain. We can open a new issue for that after this.

**Comment by GuiltyDolphin** [comment]
Fixed in #3329.
